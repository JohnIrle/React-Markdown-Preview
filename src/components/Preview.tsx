import marked from "marked";

interface PreviewProps {
  state: string;
}

export const Preview = ({ state }: PreviewProps) => {
  return (
    <div
      style={{
        border: "1px solid black",
        height: "100%",
        padding: "1rem",
      }}
      dangerouslySetInnerHTML={{
        __html: marked(state),
      }}
    />
  );
};
