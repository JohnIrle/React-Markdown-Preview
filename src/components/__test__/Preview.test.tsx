import React from "react";
import { render, screen, getByText } from "@testing-library/react";
import { Preview } from "../Preview";

const stateString = `# Try me \n ## I work \n
- lists \n - are \n - nice \n\n **Bold** \n\n *italic*\n
| one   | two   | three |\n |-------|-------|-------|\n | value  | value  | value  | \n `;

describe("Preview component", () => {
  it("renders html", () => {
    const snapshot = render(<Preview state={stateString} />);

    expect(snapshot).toMatchSnapshot();

    expect(screen.getByText(/try me/i)).toBeInTheDocument();
    expect(screen.getByText(/i work/i)).toBeInTheDocument();
    expect(screen.getByText(/bold/i)).toBeInTheDocument();
    expect(screen.getByText(/italic/i)).toBeInTheDocument();
  });
});
