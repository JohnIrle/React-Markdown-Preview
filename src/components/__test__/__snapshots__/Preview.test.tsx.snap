// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders html 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 1px solid black; height: 100%; padding: 1rem;"
      >
        <h1
          id="try-me"
        >
          Try me
        </h1>
        

        <h2
          id="i-work"
        >
          I work
        </h2>
        

        <ul>
          

          <li>
            lists 
          </li>
          

          <li>
            are 
          </li>
          

          <li>
            nice 
          </li>
          

        </ul>
        

        <p>
           
          <strong>
            Bold
          </strong>
           
        </p>
        

        <p>
           
          <em>
            italic
          </em>
        </p>
        

        <table>
          

          <thead>
            

            <tr>
              

              <th>
                one
              </th>
              

              <th>
                two
              </th>
              

              <th>
                three
              </th>
              

            </tr>
            

          </thead>
          

          <tbody>
            <tr>
              

              <td>
                value
              </td>
              

              <td>
                value
              </td>
              

              <td>
                value
              </td>
              

            </tr>
            

            <tr>
              

              <td />
              

              <td />
              

              <td />
              

            </tr>
            

          </tbody>
        </table>
        

      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 1px solid black; height: 100%; padding: 1rem;"
    >
      <h1
        id="try-me"
      >
        Try me
      </h1>
      

      <h2
        id="i-work"
      >
        I work
      </h2>
      

      <ul>
        

        <li>
          lists 
        </li>
        

        <li>
          are 
        </li>
        

        <li>
          nice 
        </li>
        

      </ul>
      

      <p>
         
        <strong>
          Bold
        </strong>
         
      </p>
      

      <p>
         
        <em>
          italic
        </em>
      </p>
      

      <table>
        

        <thead>
          

          <tr>
            

            <th>
              one
            </th>
            

            <th>
              two
            </th>
            

            <th>
              three
            </th>
            

          </tr>
          

        </thead>
        

        <tbody>
          <tr>
            

            <td>
              value
            </td>
            

            <td>
              value
            </td>
            

            <td>
              value
            </td>
            

          </tr>
          

          <tr>
            

            <td />
            

            <td />
            

            <td />
            

          </tr>
          

        </tbody>
      </table>
      

    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
